### ISSUE TITLE
luca-qr-code rejected

### ISSUE THREAD

**Issue by gitbrueck** [corona-warn-app/cwa-documentation#743]
## Avoid duplicates
* [ ] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug affects both Android **and** iOS, for specific issues / questions that apply only to one operating system, please raise them in the respective repositories:
    * [Android repository](https://github.com/corona-warn-app/cwa-app-android)
    * [iOS repository](https://github.com/corona-warn-app/cwa-app-ios)
* [ ] Bug is not already reported in another issue

## Technical details

- Device name:
- Galaxy A51
- iOS/Android version:
- Android Version 11
- App version:
- Version: 2.13.2

## Describe the bug
Luca-qr-code rejected.

## Steps to reproduce the issue
Go into a Telekom Shop and scan the luca-qr-code with the CWA.

## Expected behaviour

If I visit a Telekom Shop, I want to check in with CWA.

**Comment by Ein-Tim** [comment]
@gitbrueck 

If the luca QR code was created before 25.05.2021, it wasn't created to be comptaible with the CWA. Also, during the creation process of luca QR codes, the compatibility with CWA can be turned off.
I suggest you talk to the person which created the QR code, as we can't help you any further here.

**Comment by dsarkar** [comment]
@gitbrueck See also FAQ 
- **EN** https://www.coronawarn.app/en/faq/#check_in_luca
- **DE** https://www.coronawarn.app/de/faq/#check_in_luca

11. Can all QR codes for check-in be read immediately by the Corona-Warn-App and the luca app?

The luca-QR codes generated as of 25 May 2021 can be read (scanned) by both the Corona-Warn-App and luca app and processed for the check-in process - unless the operators have explicitly contradicted this. In the case of an incompatible luca QR code, you can contact the organizer, who will issue a new compatible luca QR code if necessary. The support of private QR codes is still not possible.

**Comment by jkrwdf** [comment]
The CWA could still recognize that the scanned QR code is a Luca QR code, and then issue a different message than currently. 

Today, for example on Android I get the generic "Kein geeigneter QR-Code - Der QR-Code wird nicht von der Corona-Warn-App unterstützt. Bitte wählen Sie einen geeigneten QR-Code". 

Much better would be however: 

"Dies ist ein Luca QR-Code, welcher jedoch die für die Corona-Warn-App notwendigen Daten noch nicht enthält"

Then the user directly knows that this is no bug of the CWA. 

For the "Luca Priavate Meetings" which are principally not supported, the message could be:

"Dies ist ein Luca QR-Code für ein privates Treffen. Solche Treffen können derzeit nicht in der Corona-Warn-App registriert werden".

**Comment by Jo-Achim** [comment]
@jkrwdf, thank you!

I believe that the proposed changes to the text are absolutely important and should be carried out in a timely manner!

Because just yesterday this article was published in "Business Insider" with the heading ["Impfnachweis, Kontaktnachverfolgung, Check-ins: Die Corona-Warn-App kann jetzt auch alle Luca-Funktionen" ("Vaccination certificate, contact tracking, check-ins: the Corona warning app can now also do all Luca functions")](https://www.businessinsider.de/politik/deutschland/impfnachweis-kontaktnachverfolgung-check-ins-die-corona-warn-app-kann-jetzt-auch-alle-luca-funktionen/).

**Comment by vaubaehn** [comment]
I support the proposal of @jkrwdf very much!

**Comment by jkrwdf** [comment]
I looked around in the Internet for some "official" Luca codes, found both old and new ones, and they all start with 

https://app.luca-app.de/webapp/<GUID>

Then I created a private Luca meeting and this QR-Code starts with 

https://app.luca-app.de/webapp/meeting/<GUID>

Looks feasible to me. 

However, the second suggestion regarding the private meetings was just a bonus. Top prio is the recognition of "https://app.luca-app.de" without CWA extension in general.

**Comment by vaubaehn** [comment]
@jkrwdf 
https://app.luca-app.de is the URI I was talking about. 

And I was wrong: not only the web app, but also the installed mobile version creates QR codes that contain that URI!
I was mixing it with the luca QR code that you can use to be checked into a venue (which seems not to be used very often)... This one does not contain any URI.

**Comment by dsarkar** [comment]
@jkrwdf @vaubaehn If you like to,  please open a wishlist item for a feature request. Thanks.

**Comment by vaubaehn** [comment]
@jkrwdf 

> If you like to, please open a wishlist item for a feature request. Thanks.

Do you like to go ahead?

**Comment by jkrwdf** [comment]
Will do

**Comment by jkrwdf** [comment]
Done: https://github.com/corona-warn-app/cwa-wishlist/issues/680

**Comment by dsarkar** [comment]
@jkrwdf ok, thanks, we will have a look at it.

**Comment by dsarkar** [comment]
@gitbrueck I suggest closing this issue and continue any related discussion on corona-warn-app/cwa-wishlist#680.

Thanks.


---
Corona-Warn-App Open Source Team

**Comment by uschindler** [comment]
Hi,
I went today through the city of Bremen and scanned every Luca code I found. In short: None of them had the CWA extension enabled. 

So what's the plan here? I am quite sure most of those codes here is Bremen were created after May 25th, but for some reason all of them have no CWA extension. They also have the letters "luca" in the center and the QR codes are quite small. All of them only have the URL like described above bat the URL is cropped after the UUID+ `#e30`. It looks like restaurants here in Bremen seem to forcefully disable CWA extensions because the generated QR code looks nicer and is easier to scan (shorter URL).

The FAQ should also contain a better explanation. 

Also why is it not possible to use the LUCA UUID and the backed server of CWA calls Luca's API (this is anonymous because the backed does the request, so no IP addresses are shared with Luca) to get the meeting/location information?

**Comment by Ein-Tim** [comment]
@uschindler 

It was and is still possible to deactivate the compatibility to CWA in the creation process of the luca QR code. 
Maybe this is the problem?

**Comment by uschindler** [comment]
That's known, but 100% of 20 tests today have disabled it. The problem needs to still be fixed, either in backend or in communication.
The CWA enabled Luca Codes are way too large and hard to scan. Maybe that's one reason. Or whatever else.
It needs to be communicated clearly and not only in some small piece of text in a FAQ.

**Comment by jkrwdf** [comment]
I do not believe that location owners intentionally switch off the CWA extension to get prettier QR codes. 

I offer another explanation for your observation which is different from "the location owners all decided against it". 

In spring when the Luca acquisition of some federal states was in the press, there was also this story: Location owners were informed that they now have a license for Luca from the state and started to generate their codes, even though at that time the public infrastructure (public health department, "Gesundheitsamt") was not even connected to Luca. Sometimes, those codes were also already posted in the venue, but most of them were then "on hold" until the location owner got the "Go". 

This might explain why in your area of living, there are nearly all Luca codes without CWA extension, although they were made publicly visible only after May 25.

**Comment by vaubaehn** [comment]
@uschindler & @jkrwdf 
In the end, I think, we will need to ask venue owners ourselves, when they created their QR codes - and when they were printed after May 25, why they decided against the CWA check-in.

> The CWA enabled Luca Codes are way too large and hard to scan.
Independently, I think it's worth to evaluate and see if we may propose changes to luca.
Given the size of the payload is fixed, scanning _may_ be more easy/reliable when
- increasing print size
- discarding any images inside the QR code (here: "luca" 😉 ) -> more space for pixels
- finding best balance for the error correction

Does luca's web interface allow for scaling print size? (I do not have access...)
Does anyone know what is the error correction level used by the web app?

**Comment by uschindler** [comment]
Definitely one venue was opened in my neighborhood beginning of October (Bremen, Horn-Lehe), so Code was created for sure after May 25th. The QR Code is a small one with Luca logo in center. No CWA payload.

I also found a place with the QR code and two images like the usual payment logos (Mastercard,...) in Window: Luca with green ✅ and then a CWA Logo next to it with red ❌. Below that the QR code.

**Comment by vaubaehn** [comment]
> I also found a place with the QR code and two images like the usual payment logos (Mastercard,...) in Window: Luca with green ✅ and then a CWA Logo next to it with red ❌. Below that the QR code.

That sounds like a conscious decision... Scannability may be one of the reasons then...

I didn't find any combined luca/CWA QR code yet. But according to another bug report it looks like, as if the full address data that is recorded from luca is inserted into the CWA code? In that case I'd propose a change to have the name only to reduce the payload size. Most important thing for the presence tracing is the CWA GUID only, afaict.

**Comment by Jo-Achim** [comment]
With the request for verification, I am also linking the article 'Corona-Warn-App: Die Rückkehr des "zahnlosen Tigers"' / 'Corona warning app: The return of the "toothless tiger"' here: https://github.com/corona-warn-app/cwa-wishlist/issues/352#issuecomment-973107941.

The Luca app may become less important in the future. But that doesn't necessarily mean giving up this topic here.
Personally, I would definitely welcome this path towards the CWA!
